### PR TITLE
Allow passing files as an artifact. 

### PR DESCRIPTION
**Pull request by holyfabi**
## **Type**
enhancement

___

## **Description**
- Introduced the ability to pass files as artifacts in the GitHub Actions workflow.
- Added new workflow steps to download and subsequently delete the specified artifacts to manage file handling within the workflow.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pullpreview.yml</strong><dd><code>Support for Passing Files as Artifacts in GitHub Actions</code>&nbsp; </dd></summary>
<hr>
      
.github/workflows/pullpreview.yml

<li>Added a new input parameter <code>FILES</code> to allow passing files as artifacts.<br> <li> Implemented steps to download and delete the specified artifact using <br>GitHub Actions.<br>

</details>
    

  </td>
  <td><a href="https://github.com/cybex-gmbh/github-workflows/pull/5/files#diff-776de6cc37d319490cdeec6f352b59a453180fee74d051d12c86a4c274687fb8">+12/-1</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: .github/workflows/pullpreview.yml [hunk 1]
```diff
@@ -54,7 +58,14 @@ jobs:
             echo "$keyValuePair" >> .env
           done
           echo "$BASIC_AUTH" > .htpasswd
-
+      - name: Download passed artifact.
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:61]: Add a space between steps.

File: .github/workflows/pullpreview.yml [hunk 2]
```diff
@@ -54,7 +58,14 @@ jobs:
...
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ inputs.FILES }}
+      - name: Delete passed artifact.
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:65]: Add a space between steps

File: .github/workflows/pullpreview.yml [hunk 3]
```diff
@@ -54,7 +58,14 @@ jobs:
...
+        uses: geekyeggo/delete-artifact@v5
+        with:
+          name: ${{ inputs.FILES }}
       - name: Boot PullPreview staging environment
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:69]: Add a space between steps

File: .github/workflows/pullpreview.yml [hunk 4]
```diff
@@ -13,6 +13,10 @@ on:
         required: false
         default: nano
         type: string
+      FILES:
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:16]: These are not really files you are passing right? It's only a name.

File: .github/workflows/pullpreview.yml [hunk 5]
```diff
@@ -13,6 +13,10 @@ on:
...
+        description: Files which can be passed as artifacts.
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:17]: Adjust description accordingly.

File: .github/workflows/pullpreview.yml [hunk 6]
```diff
@@ -54,7 +58,14 @@ jobs:
             echo "$keyValuePair" >> .env
           done
           echo "$BASIC_AUTH" > .htpasswd
-
+      - name: Download passed artifact.
+        uses: actions/download-artifact@v4
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:62]: Add a comment with a URL to the action.

File: .github/workflows/pullpreview.yml [hunk 7]
```diff
@@ -54,7 +58,14 @@ jobs:
...
+        with:
+          name: ${{ inputs.FILES }}
+      - name: Delete passed artifact.
+        uses: geekyeggo/delete-artifact@v5
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:66]: Add a comment with a URL to the action.

File: .github/workflows/pullpreview.yml (hunk 4 above)
Comment (mszulik) [inline .github/workflows/pullpreview.yml:16]: Add an example for this and also add something in the readme maybe.

**Review by gael-connan-cybex — APPROVED** [review]
looks ok, supposedly works.

File: .github/workflows/pullpreview.yml (hunk 1 above)
Comment (mszulik) [inline .github/workflows/pullpreview.yml:61]: Only download if an artifact name is passed.

File: .github/workflows/pullpreview.yml [hunk 8]
```diff
@@ -13,6 +13,10 @@ on:
         required: false
         default: nano
         type: string
+      ARTIFACT_NAME:
+        description: A Artifact which should be downloaded.
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:17]:
```suggestion
        description: The name of an Artifact which will be extracted while preserving file paths.
```

File: .github/workflows/pullpreview.yml [hunk 9]
```diff
@@ -55,6 +59,20 @@ jobs:
           done
           echo "$BASIC_AUTH" > .htpasswd
 
+      - name: Download passed artifact.
+        # https://github.com/actions/download-artifact
+        uses: actions/download-artifact@v4
+        if: ${{ inputs.ARTIFACT_NAME }}
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:65]: Write the `if` directly below the name to match the style in the other workflows.

File: .github/workflows/pullpreview.yml [hunk 10]
```diff
@@ -13,6 +13,10 @@ on:
         required: false
         default: nano
         type: string
+      ARTIFACT_NAME:
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:16]: Gael also wants a short section in the readme explaining this.

File: .github/workflows/pullpreview.yml [hunk 11]
```diff
@@ -55,6 +59,20 @@ jobs:
...
+        # https://github.com/actions/download-artifact
+        uses: actions/download-artifact@v4
+        if: ${{ inputs.ARTIFACT_NAME }}
+        with:
+          name: ${{ inputs.ARTIFACT_NAME }}
+
+      - name: Delete passed artifact.
+        # https://github.com/GeekyEggo/delete-artifact
+        uses: geekyeggo/delete-artifact@v5
+        if: ${{ inputs.ARTIFACT_NAME }}
```
Comment (mszulik) [inline .github/workflows/pullpreview.yml:72]: Did you test if this works?

**Review by mszulik — APPROVED** [review]
looks ok

**Pull request merged**